backend/src/main/java/ai/verta/modeldb/audit_log/AuditLogLocalDAORdbImpl.java

```diff
@@ -0,0 +1,35 @@
+package ai.verta.modeldb.audit_log;
+
+import ai.verta.modeldb.entities.audit_log.AuditLogLocalEntity;
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import ai.verta.modeldb.utils.ModelDBUtils;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+public class AuditLogLocalDAORdbImpl implements AuditLogLocalDAO {
+
+  private static final Logger LOGGER =
+      LogManager.getLogger(AuditLogLocalDAORdbImpl.class.getName());
+
+  public void saveAuditLogs(List<AuditLogLocalEntity> auditLogEntities) {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Transaction transaction = session.beginTransaction();
+      saveAuditLogs(session, auditLogEntities);
+      transaction.commit();
+      LOGGER.debug("Audit logged successfully");
+    } catch (Exception ex) {
+      if (ModelDBUtils.needToRetry(ex)) {
+        saveAuditLogs(auditLogEntities);
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  public void saveAuditLogs(Session session, List<AuditLogLocalEntity> auditLogEntities) {
+    auditLogEntities.forEach(session::save);
```

Comment:
Do we need to wrap this in a transaction as well to ensure the entire data set is committed as a unit of work?  The above code does this.  We should attempt to keep them the same unless there is a reason for them to differ.

Reply:
this is called from https://github.com/VertaAI/modeldb/pull/1645/files#diff-b2cc154256d0c1d31090d7d5c768bd613135f469dac3e4ae7d08b7f00220451bR20 so it is already wrapped on the transaction
```
Transaction transaction = session.beginTransaction();
      saveAuditLogs(session, auditLogEntities);
      transaction.commit();
      LOGGER.debug("Audit logged successfully");
```

Reply:
👍